docker/src/test/java/org/jclouds/docker/features/NetworkApiMockTest.java

```diff
@@ -0,0 +1,155 @@
+      } finally {
+         server.shutdown();
+      }
+   }
+
+   public void testCreateNetwork() throws Exception {
+      MockWebServer server = mockWebServer(new MockResponse().setBody(payloadFromResource("/network-creation.json")));
+      NetworkApi api = api(DockerApi.class, server.getUrl("/").toString()).getNetworkApi();
+
+      Map<String, String> options = ImmutableMap.<String, String> builder()
+              .put("com.docker.network.bridge.default_bridge", "true")
+              .put("com.docker.network.bridge.enable_icc", "true")
+              .put("com.docker.network.bridge.enable_ip_masquerade", "true")
+              .put("com.docker.network.bridge.host_binding_ipv4", "0.0.0.0")
+              .put("com.docker.network.bridge.name", "docker0")
+              .put("com.docker.network.driver.mtu", "1500")
+              .build();
+
+      Network network = Network.create(
+              "isolated_nw", // Name
+              null, // Id
+              "bridge", // Driver
+              null, // Scope
+              Network.IPAM.create(
+                      "default", // driver
+                      ImmutableList.of(Network.IPAM.Config.create("172.17.0.0/16", null, null)) // config
+              ),
+              ImmutableMap.of("39b69226f9d79f5634485fb236a23b2fe4e96a0a94128390a7fbbcc167065867",
+                      Network.Details.create(
+                              "ed2419a97c1d9954d05b46e462e7002ea552f216e9b136b80a7db8d98b442eda", //endpointId
+                              "02:42:ac:11:00:02", // MAC
+                              "172.17.0.2/16", // ipv4address
+                              "" // ipv6address
+                      )
+              ),
+              options);
+
+      try {
+         Network Network = api.createNetwork(network);
+         assertSent(server, "POST", "/networks/create");
+         assertNotNull(Network);
+         assertThat(Network.id()).isEqualTo("22be93d5babb089c5aab8dbc369042fad48ff791584ca2da2100db837a1c7c30");
+      } finally {
+         server.shutdown();
+      }
+   }
+
+   public void testRemoveNetwork() throws Exception {
+      MockWebServer server = mockWebServer(new MockResponse().setResponseCode(204));
+      NetworkApi api = api(DockerApi.class, server.getUrl("/").toString()).getNetworkApi();
+      String networkId = "6d35806c1bd2b25cd92bba2d2c2c5169dc2156f53ab45c2b62d76e2d2fee14a9";
+
+      try {
+         api.removeNetwork(networkId);
+         assertSent(server, "DELETE", "/networks/" + networkId);
+
+      } finally {
+         server.shutdown();
+      }
+   }
+
+   public void testConnectContainerToNetwork() throws Exception {
+      MockWebServer server = mockWebServer(new MockResponse().setResponseCode(200));
+      NetworkApi api = api(DockerApi.class, server.getUrl("/").toString()).getNetworkApi();
+      try {
+         api.connectContainerToNetwork("1", "containerName");
+         assertSent(server, "POST", "/networks/1/connect");
```

Comment:
I think I was using `@Produces` annotation in a wrong way as I had to use `@Headers` for `Content-Type` so I think it is now better. Sorry for the confusion!